dali/operators/segmentation/random_object_bbox.cc

```diff
@@ -0,0 +1,458 @@
+using kernels::connected_components::LabelConnectedRegions;
+using kernels::label_bbox::GetLabelBoundingBoxes;
+
+DALI_SCHEMA(segmentation__RandomObjectBBox)
+  .DocStr(R"(Randomly selects an object from a mask and returns its bounding box.
+
+This operator takes a labeled segmentation map as its input. With probability ``foreground_prob``
+it randomly selects a label (uniformly or according to the distribution given as ``class_weights``),
+extracts connected blobs of pixels with the selected label and randomly selects one of the blobs.
+The blobs may be further filtered according to ``k_largest`` and ``threshold``.
+The output is a bounding box of the selected blob in one of the formats described in ``format``.
+
+With probability 1-foreground_prob, the entire area of the input is returned.)")
+  .NumInput(1)
+  .OutputFn([](const OpSpec& spec) {
+    int separate_corners = spec.GetArgument<string>("format") != "box";
+    int output_class = spec.GetArgument<bool>("output_class");
+    return 1 + separate_corners + output_class;
+  })
+  .AddOptionalArg("ignore_class", R"(If True, all objects are picked with equal probability,
+regardless of the class they belong to. Otherwise, a class is picked first and then an object is
+randomly selected from this class.
+
+This argument is incompatible with ``classes``, ``class_weights`` or ``output_class``.
+
+.. note::
+  This flag only affects the probability with which blobs are selected. It does not cause
+  blobs of different classes to be merged.)", false)
+  .AddOptionalArg("output_class", R"(If True, an additional output is produced which contains the
+label of the class to which the selected box belongs, or background label if the selected box
+is not an object bounding box.
+
+The output may not be an object bounding box when any of the following conditions occur:
+  - the sample was randomly (according to ``foreground_prob``) chosen not be be a foreground one
+  - the sample contained no foreground objects
+  - no bounding box met the required size threshold.)", false)
+  .AddOptionalArg("foreground_prob", "Probability of selecting a foreground bounding box.", 1.0f,
+    true)
+  .AddOptionalArg<vector<int>>("classes", R"(List of labels considered as foreground.
+
+If left unspecified, all labels not equal to ``background`` are considered foreground.)",
+    nullptr, true)
+  .AddOptionalArg("background", R"(Background label.
+
+If left unspecified, it's either 0 or any value not in ``classes``.)", 0, true)
+  .AddOptionalArg<vector<float>>("class_weights", R"(Relative probabilities of foreground classes.
+
+Each value corresponds to a class label in ``classes``. If ``classes`` are not specified,
+consecutive 1-based labels are assigned.
+
+The sum of the weights doesn't have to be equal to 1 - if it isn't the weights will be
+normalized .)", nullptr, true)
+  .AddOptionalArg<int>("k_largest", R"(If specified, the boxes are sorted by decreasing volume
+and only ``k_largest`` are considered.)",
```

Comment:
Total `k-largest` or per class?

Reply:
That depends on ``ignore_class`` - I don't know if it makes it better if we write it here or not.